src/main/java/org/broadinstitute/consent/http/resources/TDRResource.java

```diff
@@ -64,4 +86,45 @@ public Response getDatasetByIdentifier(@Auth AuthUser authUser, @PathParam("iden
             return createExceptionResponse(e);
         }
     }
+
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/dar/draft")
+    @PermitAll
+    public Response createDraftDataAccessRequest(
+        @Auth AuthUser authUser,
+        @Context UriInfo info,
+        @QueryParam("identifiers") String identifiers,
+        @QueryParam("projectTitle") String projectTitle) {
+      try {
+        // Ensure that the user is a registered DUOS and SAM user
+        User user = userService.findOrCreateUser(authUser);
+        if (Objects.isNull(identifiers) || identifiers.isBlank()) {
+          throw new BadRequestException("No dataset identifiers were provided");
+        } else {
+          List<String> identifierList = Arrays.stream(identifiers.split(","))
+                  .map(String::trim)
+                  .toList();
+          List<Integer> datasetIds = tdrService.getDatasetIdsByIdentifier(identifierList);
+        // Check that we were able to find a dataset id for all identifiers provided
+        if (identifierList.size() != datasetIds.size()) {
+            throw new BadRequestException("Invalid dataset identifiers were provided.");
+        }
```

Comment:
I think a more appropriate error here would be a not found exception for the identifiers that we could not find as opposed to a BRE. It would be nice to let the client know which identifiers were NOT found so they can rectify their request.

Reply:
Refactored the endpoint logic to produce a list of the identifiers that did not match to a dataset when the number of datasets do not match the number of identifiers provided. Also added a specific test case for the NFE.